pkgs/by-name/qb/qbittorrent/package.nix

```diff
@@ -0,0 +1,98 @@
+{
+  apple-sdk_11,
+  boost,
+  cmake,
+  dbus,
+  fetchFromGitHub,
+  guiSupport ? true,
+  hicolor-icon-theme,
+  lib,
+  libtorrent-rasterbar,
+  nix-update-script,
+  openssl,
+  pkg-config,
+  python3,
+  qt6,
+  stdenv,
+  trackerSearch ? true,
+  webuiSupport ? true,
+  wrapGAppsHook3,
+  zlib,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "qbittorrent" + lib.optionalString (!guiSupport) "-nox";
+  version = "5.0.2";
+
+  src = fetchFromGitHub {
+    owner = "qbittorrent";
+    repo = "qBittorrent";
+    rev = "release-${finalAttrs.version}";
+    hash = "sha256-JIURzAkVNYjHAx8yY0WaKNK4N/z9ndqjk0EXkJH9AzY=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    wrapGAppsHook3
+    qt6.wrapQtAppsHook
+  ];
+
+  buildInputs =
+    [
+      boost
+      hicolor-icon-theme
```

Comment:
```suggestion
```

I don't think `hicolor-icon-theme` is needed, neither [upstream][1] nor [Debian][2] include it.

[1]: https://github.com/qbittorrent/qBittorrent/blob/master/INSTALL
[2]: https://packages.debian.org/source/sid/qbittorrent

The `openssl` and `zlib` buildInputs addition should be fine, although they're already propagate by qtbase.

Reply:
Is a fallback theme too much?
https://gitlab.archlinux.org/archlinux/packaging/packages/qbittorrent/-/blob/main/PKGBUILD#L49

Reply:
We don't need that at least for qbittorrent-nox and darwin.

Reply:
And I think theming is a system level thing, not package level, unless the package explicitly need it.

Reply:
What would you do in the case of qbittorrent-enhanced?
The name of the binary is qbittorrent as well after compiling.
Would you just rename postInstall and mainProgram to use just qbittorrent instead of qbittorrent-enhanced?